sweep/fee_bumper.go

```diff
@@ -76,4 +92,436 @@ type BumpResult struct {
+
+	// Publish the sweeping tx with customized label. If the publish fails,
+	// this error will be saved in the `BumpResult` and it will be removed
+	// from being monitored.
+	err = t.cfg.Wallet.PublishTransaction(
+		tx, labels.MakeLabel(labels.LabelTypeSweepTransaction, nil),
+	)
+	if err != nil {
+		// NOTE: we decide to attach this error to the result instead
+		// of returning it here because by the time the tx reaches
+		// here, it should have passed the mempool acceptance check. If
+		// it still fails to be broadcast, it's likely a non-RBF
+		// related errors happened. So we send this error back to the
+		// caller so that it can handle it properly.
+		//
+		// TODO(yy): find out which input is causing the failure.
+		log.Errorf("Failed to publish tx %v: %v", txid, err)
+	}
+
+	result := &BumpResult{
+		CurrentTx: record.tx,
+		Fee:       record.fee,
+		FeeRate:   record.feeFunction.FeeRate(),
+		Err:       err,
+		requestID: requestID,
+	}
+
+	return result, nil
+}
+
+// notifyAndRemove sends the result to the resultChan specified by the
+// requestID. This channel is expected to be read by the caller. If the result
+// contains a non-nil error, or the tx is confirmed, the record will be removed
+// from the maps.
+func (t *TxPublisher) notifyAndRemove(result *BumpResult) {
+	id := result.requestID
+	resultChan, ok := t.resultChans.Load(id)
+	if !ok {
+		log.Errorf("Result chan for id=%v not found", id)
+		return
+	}
+
+	log.Debugf("Sending result for requestID=%v", id)
+
+	select {
+	// TODO(yy): Add timeout in case it's blocking?
+	case resultChan <- result:
+
+	case <-t.quit:
+		log.Debug("Fee bumper stopped")
+	}
+
+	// Remove the record from the maps if there's an error. This means this
+	// tx has failed its broadcast and cannot be retried. There are two
+	// cases,
+	// - when the budget cannot cover the fee.
+	// - when a non-RBF related error occurs.
+	if result.Err != nil {
+		log.Errorf("Removing monitor record=%v due to err: %v",
+			id, result.Err)
+		t.resultChans.Delete(id)
+		t.records.Delete(id)
+	}
+
+	// Remove the record is the tx is confirmed.
+	if result.Confirmed {
+		log.Infof("Removing confirmed monitor record=%v", id)
+		t.resultChans.Delete(id)
+		t.records.Delete(id)
+	}
+}
+
+// monitorRecord is used to keep track of the tx being monitored by the
+// publisher internally.
+type monitorRecord struct {
+	// tx is the tx being monitored.
+	tx *wire.MsgTx
+
+	// set is the input set used to create the tx. It's supplied by the
+	// caller.
+	set InputSet
+
+	// changeScript is the script to send the change output to.
+	changeScript []byte
+
+	// confEvent is the subscription to the confirmation event of the tx.
+	confEvent *chainntnfs.ConfirmationEvent
+
+	// feeFunction is the fee bumping algorithm used by the publisher.
+	feeFunction FeeFunction
+
+	// fee is the fee paid by the tx.
+	fee btcutil.Amount
+}
+
+// TODO(yy): temp, remove this placeholder once the TestMempoolAccept PR is
```

Comment:
Q: Which PR are you referring to? I think this is a prerequisite before we merge this PR ?